### ISSUE TITLE
After creating the projection, NOT_FOUND_COLUMN_IN_BLOCK exception was encountered when inserting data from parquet file.

### ISSUE THREAD

**Issue by fsgdgsdfgsgd** [ClickHouse/ClickHouse#40514]
Hi guys,
We are trying to optimize some queries using projection. However, when we set up the GROUP BY projection, we found that we encountered this error when inserting data.
```
Error: Received exception from server (version 22.8.1):
Code: 10. DB::Exception: Received from localhost:9000. 
DB::Exception: Not found column toStartOfDay(toDateTime(date)) in block. 
There are only columns: date, market, uniqExact(if(equals(isuser, 1), userid, NULL)). (NOT_FOUND_COLUMN_IN_BLOCK)
(query: INSERT INTO table (A, B, C ......) FORMAT Parquet)
```
Because of some confidentiality issues, I used some placeholders.
Our projection is like this:
```
ALTER TABLE table ADD PROJECTION test_projection
(
	SELECT
		toStartOfDay(toDateTime(date)) AS Day,
		date,
	        market,
		count(DISTINCT IF(isuser = 1, userid, NULL))
	GROUP BY Day, date, market
);
```
But I can't reproduce this error myself, this error does appear stably after adding projection. Does anyone know what could be the reason for this?

**Comment by fsgdgsdfgsgd** [comment]
Hi guys,

We found a way that can reproduce this error. We think this might be a bug.
Here is the code:
```
1. Create a table
CREATE TABLE default.test_table
(
	bool_value UInt8,
	zero_integer_value Int32,
	integer_value Int32,
	float_value Float32,
	double_value Float64,
	datetime_value DateTime,
	decimal_value Decimal(28,8),
	string_value String,
	array_value Array(String)
)
engine = ReplicatedMergeTree('/Project/Test_Env/test_table/shard_prod', '{host}')
partition by toDate(datetime_value)
order by bool_value;

2. Add projection
ALTER TABLE default.test_table ADD PROJECTION test_projection
(
	SELECT
		toStartOfDay(toDateTime(datetime_value)) AS Day,
		datetime_value,
		float_value,
		COUNT(DISTINCT IF(zero_integer_value = 1, string_value, NULL))
	GROUP BY Day, datetime_value, float_value
);

3. Insert data
cat basic_column.parquet | clickhouse-client --query="INSERT INTO default.test_table (bool_value, zero_integer_value, integer_value, float_value, double_value, datetime_value, decimal_value, string_value, array_value) FORMAT Parquet"
```
I put the file 'basic_column.parquet' in the attachment.
[basic_column.zip](https://github.com/ClickHouse/ClickHouse/files/9401982/basic_column.zip)

**Comment by amosbird** [comment]
This is similar to the issue that https://github.com/ClickHouse/ClickHouse/pull/49709 is going to fix. Will fix this one after #49709 is merged.
